maro/simulator/scenarios/vm_scheduling/business_engine.py

```diff
@@ -129,29 +145,47 @@ def _init_data(self):
         if (not os.path.exists(vm_table_data_path)) or (not os.path.exists(cpu_readings_data_path)):
             self._download_processed_data()
 
+    def _cal_pm_amount(self) -> int:
+        amount: int = 0
+        for pm_type in self._config.PM:
+            amount += pm_type["amount"]
+
+        return amount
+
     def _init_pms(self):
         """Initialize the physical machines based on the config setting. The PM id starts from 0."""
-        self._pm_cpu_cores_capacity: int = self._config.PM.CPU
-        self._pm_memory_capacity: int = self._config.PM.MEMORY
-
         # TODO: Improve the scalability. Like the use of multiple PM sets.
         self._machines = self._frame.pms
-        for pm_id in range(self._pm_amount):
-            pm = self._machines[pm_id]
-            pm.set_init_state(
-                id=pm_id,
-                cpu_cores_capacity=self._pm_cpu_cores_capacity,
-                memory_capacity=self._pm_memory_capacity
-            )
+        # PM type dictionary.
+        self._pm_type_dict: dict = {}
+        pm_id = 0
+        for pm_type in self._config.PM:
+            amount = pm_type["amount"]
+            self._pm_type_dict[pm_type["PM_type"]] = pm_type
+            while amount > 0:
+                pm = self._machines[pm_id]
+                pm.set_init_state(
+                    id=pm_id,
+                    cpu_cores_capacity=pm_type["CPU"],
+                    memory_capacity=pm_type["memory"],
+                    pm_type=pm_type["PM_type"],
```

Comment:
It's a string or int here?

Reply:
After loading from config, the key is string `["PM_type"]`, the value is `int`